libmultilabel/model.py

```diff
@@ -77,39 +94,52 @@ def test_epoch_end(self, step_outputs):
         return eval_metric
 
     def evaluate(self, step_outputs, split):
-        eval_metric = MultiLabelMetrics(self.config)
+        eval_metric = MultiLabelMetrics(self.metric_threshold, self.monitor_metrics)
         for step_output in step_outputs:
             eval_metric.add_values(y_pred=step_output['pred_scores'],
                                    y_true=step_output['target'])
         metric_dict = eval_metric.get_metric_dict()
         self.log_dict(metric_dict)
-        dump_log(config=self.config, metrics=metric_dict, split=split)
+        dump_log(metrics=metric_dict, split=split, log_path=self.log_path)
 
         self.print(f'\n====== {split.upper()} dataset evaluation result =======')
         self.print(eval_metric)
         return eval_metric
 
     def print(self, string):
-        if not self.config.get('silent', False):
+        if not self.silent:
             if not self.trainer or self.trainer.is_global_zero:
                 print(string)
 
 
 class Model(MultiLabelModel):
-    def __init__(self, config, word_dict=None, classes=None):
-        super().__init__(config)
+    def __init__(
+        self,
+        device,
+        model_name,
+        classes=None,
+        word_dict=None,
+        init_weight=None,
+        log_path=None,
+        **kwargs
+    ):
+        super().__init__(log_path=log_path, **kwargs)
         self.save_hyperparameters()
 
         self.word_dict = word_dict
         self.classes = classes
-        self.config.num_classes = len(self.classes)
+        self.num_classes = len(self.classes)
 
         embed_vecs = self.word_dict.vectors
-        self.network = getattr(networks, self.config.model_name)(
-            self.config, embed_vecs)
-
-        if config.init_weight is not None:
-            init_weight = networks.get_init_weight_func(self.config)
+        self.network = getattr(networks, model_name)(
+            embed_vecs=embed_vecs,
+            num_classes=self.num_classes,
+            **kwargs
```

Comment:
I left`**kwargs`here because there could be different network parameters for different models. To make it easy to add new models without maintaining the parameter map, I prefer` **kwargs` here. What do you think?

Reply:
Maybe we can construct the network outside then pass it to the model.
Just like the example provided in Pytorch Lightning's document: https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#inference-in-production
But we can just keep it for now.